csharp/HOLMS.Platform/HOLMS.Platform/Application.Client/ApplicationClient.cs

```diff
@@ -141,6 +141,19 @@ public class ApplicationClient : IDisposable, IApplicationClient {
             _clientId = oauth2ClientId;
         }
 
+        /// <summary>
+        /// Used to update the server connection profile. Disposes existing session.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="oauth2ClientId"></param>
+        public virtual void UpdateConfiguration(IApplicationClientConfig config, string oauth2ClientId) {
```

Comment:
In general I strongly prefer having only one way to do things in the code. Adding this means you could either throw out the old reference like I've described, or call this method. This will confuse future devs.